### ISSUE TITLE
ACHDirectDebit not coming down with paymentMethods

### ISSUE THREAD

**Issue by lpuentes19** [Adyen/adyen-ios#1114]
**Describe the challenge**
When we call `AdyenSession.initialize` for a session where there is no amount to pay, no invoice, etc. the amount is $0.00, the response gives us back a `session` object which has a `sessionContext` property which holds a `paymentMethods` property. This `paymentMethods` array contains a CC and ACH payment method as "regular" payment methods when there is an amount to pay (amount greater than $0.00). But it does not contain ACH when there is no amount to pay and therefore blocks us from opening up the ACH form. We were hoping to allow the user to use the ACHDirectDebitComponent to add and save their payment method without having to make a payment. Is there anyway to get the `paymentMethods` array mentioned above to contain ACH payment method when there is no amount to pay? CC payment method shows no matter the scenario.

I did see this in the documentation: https://docs.adyen.com/online-payments/ios/components

I noticed on step 1 where we provide an amount, it says the amount is used to filter the list of available payment methods to your shopper. Maybe Adyen is filtering out ACH payment method when the amount is $0.00? If so, why is this? I've met with the BE devs and they have assured me they're not manipulating anything and just passing the data to and from Adyen.

Here is a snippet of the code that I'm referring to:

```
if let sessionId = sessionId, let sessionData = sessionData, let adyenContext = try? AdyenContext(apiContext: APIContext(environment: environment, clientKey: clientKey), payment: nil) {
            let configuration = AdyenSession.Configuration(sessionIdentifier: sessionId, initialSessionData: sessionData, context: adyenContext)
            
            AdyenSession.initialize(with: configuration, delegate: self, presentationDelegate: self) { [weak viewController] result in
                switch result {
                case .success(let session):
                    // ACH Component configuration
                    let style = adyenFormComponentStyle()
                    var achComponentConfig = ACHDirectDebitComponent.Configuration()
                    achComponentConfig.style = style
                    
                    // Create ACH Component - *Does not make it past this guard statement*
                    guard let paymentMethods = session.sessionContext.paymentMethods,
                            let paymentMethod = paymentMethods.paymentMethod(ofType: ACHDirectDebitPaymentMethod.self) else { return }

                    let achComponent = ACHDirectDebitComponent(paymentMethod: paymentMethod, context: adyenContext, configuration: achComponentConfig)
achComponent.delegate = session
                case .failure(let error):
                    print(error)
                }
            }
        }
```

**Comment by descorp** [comment]
Hey @lpuentes19 

Is this behaviour on both LIVE and TEST?

>  Maybe Adyen is filtering out ACH payment method when the amount is $0.00?

This is a fair assumption.
I would advise contacting [support team](https://www.adyen.help/hc/en-us/requests/new?ticket_form_id=360000705420) - they will help you to enable recurring ACH payments on your CA

**Comment by erenbesel** [comment]
Hey @lpuentes19 
In addition to @descorp's suggestion to enable recurring payments, there indeed is a minimum amount required for ACH Direct Debit to be available as a regular payment method, which is $1.00.

**Comment by erenbesel** [comment]
Closing this hoping it was resolved and due to inactivity.
